app/src/ui/history/commit-list-item.tsx

```diff
@@ -1,14 +1,22 @@
 import * as React from 'react'
 import { Commit } from '../../models/commit'
+import { Repository } from '../../models/repository'
 import { IAvatarUser } from '../../models/avatar'
 import { RichText } from '../lib/rich-text'
 import { Avatar } from '../lib/avatar'
 import { RelativeTime } from '../relative-time'
+import { getDotComAPIEndpoint } from '../../lib/api'
+import { clipboard } from 'electron'
+import { showContextualMenu, IMenuItem } from '../main-process-proxy'
 
 interface ICommitProps {
+  readonly repository: Repository | null
```

Comment:
> When would the repository be null? 🤔

The "welcome wizard" flow, which Isn't Great but we can :boot: that to whatever we end up doing with #2141 .